pkg/crc/cluster/cluster.go

```diff
@@ -73,24 +91,31 @@ func AddPullSecretInTheCluster(ocConfig oc.Config, pullSec *PullSecret) error {
 		fmt.Sprintf(`'{"data":{".dockerconfigjson":"%s"}}'`, base64OfPullSec),
 		"-n", "openshift-config", "--type", "merge"}
 
-	if err := WaitForOpenshiftResource(ocConfig, "secret"); err != nil {
-		return err
-	}
 	_, stderr, err := ocConfig.RunOcCommandPrivate(cmdArgs...)
 	if err != nil {
 		return fmt.Errorf("Failed to add Pull secret %v: %s", err, stderr)
 	}
 	return nil
 }
 
-func UpdateClusterID(ocConfig oc.Config) error {
+func EnsureClusterIDIsNotEmpty(ocConfig oc.Config) error {
```

Comment:
Why we are adding thee check, this is not something user facing and we automatic add it as part of code, I am not sure what benefit it provide, if it is not set that mean our code which set this is buggy and need to be fixed.

Reply:
I would like to remove most of the branches in the `machine.Start` code. If the creation code could be the same as re-start code, it would be easier to follow. 

It also ensures after each start that the clusterID, the pull secret, etc. are correctly in place.

Reply:
this rationale belongs in the commit log

Reply:
added

Reply:
>  I am not sure what benefit it provide, if it is not set that mean our code which set this is buggy and need to be fixed.

It would help to recover in case there was a transient cluster failure when the cluster was created, or if the user pressed ctrl+c during the initial cluster creation, ...